lib/compilers/sail.ts

```diff
@@ -0,0 +1,177 @@
+//       documentation and/or other materials provided with the distribution.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+// LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+// CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+// SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+// INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+// CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+// POSSIBILITY OF SUCH DAMAGE.
+
+import fs from 'fs-extra';
+import path from 'path';
+
+import type {PreliminaryCompilerInfo} from '../../types/compiler.interfaces.js';
+import type {ParseFiltersAndOutputOptions} from '../../types/features/filters.interfaces.js';
+import {BaseCompiler} from '../base-compiler.js';
+import {CompilationEnvironment} from '../compilation-env.js';
+import { CompilationResult, ExecutionOptionsWithEnv } from '../../types/compilation/compilation.interfaces.js';
+import { SelectedLibraryVersion } from '../../types/libraries/libraries.interfaces.js';
+
+export class SailCompiler extends BaseCompiler {
+    static get key() {
+        return 'sail';
+    }
+
+    constructor(info: PreliminaryCompilerInfo, env: CompilationEnvironment) {
+        super(info, env);
+
+        this.outputFilebase = 'model';
+    }
+
+    override optionsForFilter(filters: ParseFiltersAndOutputOptions, outputFilename: any) {
+        // By default this adds C compiler options (-g etc).
+
+        // Add a fake option so we know to compile to binary in `runCompiler()`.
+        // This is how most other compilers work (with something like `-S`), and
+        // the `BaseCompiler` isn't really set up to allow anything else.
+        if (filters.binary) {
+            return ["--link-to-binary"];
+        }
+        return [];
+    }
+
+    override async runCompiler(
+        compiler: string,
+        options: string[],
+        inputFilename: string,
+        execOptions: ExecutionOptionsWithEnv,
+        filters?: ParseFiltersAndOutputOptions,
+    ): Promise<CompilationResult> {
+        if (!execOptions) {
+            execOptions = this.getDefaultExecOptions();
+        }
+
+        const tmpDir = path.dirname(inputFilename);
+
+        if (!execOptions.customCwd) {
+            execOptions.customCwd = tmpDir;
+        }
+
+        // We can't use `filters` to decide whether to compile to binary
+        // since it isn't always set (e.g. when this is called from
+        // buildExecutable()). Instead we use a fake command line option.
```

Comment:
This may now be set OK - I merged in the latest code into your branch, maybe take another look? Thanks

Reply:
Will do.